node/src/index.ts

```diff
@@ -92,33 +92,53 @@ export class Table {
   }
 
   /**
-     * Create a search query to find the nearest neighbors of the given query vector.
-     * @param queryVector The query vector.
-     */
-  search (queryVector: number[]): Query {
+    * Creates a search query to find the nearest neighbors of the given query vector.
+    * @param queryVector The query vector.
+    */
+  search (queryVector: number[]): Query
+  /**
+   * Creates a search query to find the nearest neighbors of the given search term
+   * @param query The query search term
+   * @param embeddings An embedding function used to vectorize the query vector
+   */
+  search<T> (query: T, embeddings: EmbeddingFunction<T>): Query
+  search<T> (query: T | number[], embeddings?: EmbeddingFunction<T>): Query {
```

Comment:
Can the Table just hold a reference to the embedding function? I'm not sure what the advantage here just saving one function call.